include/boost/beast/core/string.hpp

```diff
@@ -10,21 +10,45 @@
 #ifndef BOOST_BEAST_STRING_HPP
 #define BOOST_BEAST_STRING_HPP
 
+#include <boost/asio/detail/string_view.hpp>
+
+#ifndef BOOST_ASIO_NO_STRING_VIEW
+# if defined(BOOST_ASIO_HAS_STD_STRING_VIEW)
+#  define BOOST_ASIO_HAS_STRING_VIEW 0
+# else
+#  define BOOST_ASIO_HAS_STRING_VIEW 1
```

Comment:
I don't think we are allowed to adjust these macros. The user needs to adjust them not our library.

Reply:
I was simplified things a bit more.